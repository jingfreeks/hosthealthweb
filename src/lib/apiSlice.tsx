import { setCredentials, setLogout } from "@/slice/auth";
import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";

const baseQuery = fetchBaseQuery({
  baseUrl: process.env.DEV_BASE_URL,
  credentials: "include",
  prepareHeaders: (headers, { getState }: any) => {
    const token = getState().auth.token;
    if (token) {
      headers.set("authorization", `Bearer ${token}`);
    }
    return headers;
  },
});

const baseQueryWithAuth = async (
  args: string,
  api: any,
  extraOptions: string
) => {
  let result: any = await baseQuery(args, api, extraOptions);
  if (result?.error?.originalStatus === 403) {
    const refereshResult = await baseQuery("/refresh", api, extraOptions);

    if (refereshResult?.data) {
      const user = api.getState().auth.user;
      //store the new token
      api.dispatch(setCredentials({ ...refereshResult.data, user }));
      //retry the original query with new access token
      result = await baseQuery(args, api, extraOptions);
    } else {
      api.dispatch(setLogout());
    }
  }
  return result;
};

export const apiSlice = createApi<any, any>({
  baseQuery: baseQueryWithAuth,
  reducerPath: "api",
  endpoints: (builder) => ({}),
  tagTypes: ["City", "Jobs", "MyJobs", "JobDetails", "Profile"] as any,
});
