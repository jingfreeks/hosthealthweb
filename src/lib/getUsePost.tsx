import React from "react";

export default async function getUserPosts(userId: string) {
  const response = await fetch(
    `https://jsonplaceholder.typicode.com/posts?userId=${userId}`,
    { next: { revalidate: 60 } }
  );

  if (!response.ok) undefined;
  return response.json();
}
