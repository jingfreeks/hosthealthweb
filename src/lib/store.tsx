import { configureStore, combineReducers } from "@reduxjs/toolkit";
import authReducer from "@/slice/auth";
import {apiSlice} from './apiSlice';

const rootReducer = combineReducers({
  [apiSlice.reducerPath]: apiSlice.reducer,
  auth: authReducer,
});
export const setupStore= (preloadedState?: Partial<RootState>) => {
  return configureStore({
    reducer: rootReducer,
    middleware: getDefaultMiddleware =>
        getDefaultMiddleware({ serializableCheck: false}).concat(apiSlice.middleware),
      devTools: true,
    preloadedState
  });
};

// Infer the type of makeStore
export type AppStore = ReturnType<typeof setupStore>;
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof rootReducer>
export type AppDispatch = AppStore["dispatch"];
