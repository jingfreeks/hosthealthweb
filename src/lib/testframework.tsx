/* eslint-disable @typescript-eslint/no-explicit-any */
import React,{PropsWithChildren} from 'react';
import {render} from '@testing-library/react';
import type {RenderOptions} from '@testing-library/react';
import {Provider} from 'react-redux';
import type {AppStore, RootState} from './store';
import {setupStore} from './store'
// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  // eslint-disable-next-line @typescript-eslint/ban-types
  function Wrapper({children}: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
