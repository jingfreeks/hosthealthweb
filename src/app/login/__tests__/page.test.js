import { renderWithProviders } from "../../../lib/testframework";
import Page from "../page";

describe("Login Screen Page", () => {
  it("renders Login Screen Page to match snaptshots", async () => {
    const { container } = renderWithProviders(<Page />);
    expect(container).toMatchSnapshot();
  });
});
