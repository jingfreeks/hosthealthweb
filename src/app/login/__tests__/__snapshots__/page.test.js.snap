// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen Page renders Login Screen Page to match snaptshots 1`] = `
<div>
  <div
    class="h-screen w-screen flex justify-center items-center bg-slate-100"
  >
    <div
      class="sm:shadowo-xl px-8 pb-8 pt-12 sm:bg-white rounded-xl space-y-12"
    >
      <h1
        class="font-semibold text-2xl"
      >
        Login
      </h1>
      <form
        class="space-y-12 w[400px]"
        data-testid="RegisterMainForm"
      >
        <div
          class="grid w-full max-w-sm items-center gap-1.5"
        >
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            for="username"
          >
            Username
          </label>
          <input
            class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
            id="username"
            name="username"
            type="username"
          />
          <p />
        </div>
        <div
          class="grid w-full max-w-sm items-center gap-1.5"
        >
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            for="password"
          >
            Password
          </label>
          <input
            class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
            id="password"
            name="password"
            type="password"
          />
          <p />
        </div>
        <div
          class="w-full"
        >
          <button
            class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-11 rounded-md px-8 w-full"
          >
            Login
          </button>
        </div>
      </form>
      <p
        class="text-center"
      >
        Need to create an account?
        <a
          class="text-indigo-500 hover:underline"
          href="/register"
        >
          Create Account
        </a>
      </p>
    </div>
  </div>
</div>
`;
